File: modified_scripts/rust/binary_search/gemma_2b.rs
```rust
fn binary_search(target: u32, arr: &[u32]) -> i32 {
    let low = 0;
    let high = arr.len();

    while low <= high {
        let mid = (low + high) / 2;
        if arr[mid] == target {
            return mid;
        } else if arr[mid] < target {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }

    -1
}
```

File: modified_scripts/javascript/binary_search/gemma_2b.js

function binarySearch(arr, target) {
  let low = 0;
  let high = arr.length;

  while (low <= high) {
    const mid = (low + high) / 2;
    if (arr[mid] === target) {
      return mid;
    } else if (arr[mid] < target) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }

  return -1;
}
